independent-projects/arc/tests/src/test/java/io/quarkus/arc/test/event/select/EventSelectTest.java

```diff
@@ -22,67 +22,56 @@ public class EventSelectTest {
 
     @Test
     public <T> void testEventSelectThrowsExceptionIfEventTypeHasTypeVariable() {
-        try {
-            SecuritySensor sensor = Arc.container().select(SecuritySensor.class).get();
-            sensor.securityEvent.select(new TypeLiteral<SecurityEvent_Illegal<T>>() {
-            });
-            Assertions.fail("Event#select should throw IllegalArgumentException if the event uses type variable");
-        } catch (IllegalArgumentException iae) {
-            // expected
-        }
+        SecuritySensor sensor = Arc.container().select(SecuritySensor.class).get();
+        TypeLiteral<SecurityEvent_Illegal<T>> typeLiteral = new TypeLiteral<SecurityEvent_Illegal<T>>() {
```

Comment:
A beer to whoever can explain to me why declaring this `TypeLiteral` directly inside the lambda leads to the type variable not being recognized correctly :)

Reply:
:shrug:

Reply:
Same here. No beer for me.